cmd/server.go

```diff
@@ -62,9 +65,12 @@ var serverCmd = &cobra.Command{
 		}
 
 		server := &http.Server{
-			Addr:         flagListenAddr,
-			ReadTimeout:  5 * time.Second,
-			WriteTimeout: 5 * time.Second,
+			Addr:        flagListenAddr,
+			ReadTimeout: 5 * time.Second,
+			// WriteTimeout is specifically set to 0 in order to disable the timeout. Downloading large binaries
+			// through the pull-through cache from slow upstream servers might exceed any timeout set here.
+			// This is definitely not ideal, as it impacts robustness.
+			WriteTimeout: 0,
```

Comment:
We could also set the `WriteTimeout` to a value big enough for most cases. Like `time.Second * 120` for example.

Reply:
That's a sensible idea, thanks! Added the flag in 811062f50b981f0a36d1623b707b5c0073cb6b6c

Reply:
That's a sensible idea, thanks! Added the flag in 811062f50b981f0a36d1623b707b5c0073cb6b6c